src/lib.rs

```diff
@@ -1,7 +1,76 @@
-#![feature(test)]
+//! # Hades252
+//! 
+//! Implementation of Hades252 over the Ristretto Scalar field.
+//! 
+//! *Unstable* : No guarantees can be made regarding the API stability.
+//! 
+//! ## Parameters
+//! 
+//! - p = `2^252 + 27742317777372353535851937790883648493`
+//! 
+//! - Security level is 126 bits
+//! 
+//! - width = 9
+//! 
+//! - Number of full rounds = 8 . There are four full rounds at the beginning and four full rounds at the end
+//! 
+//! - Merkle tree arity : 8 to 1
+//! 
+//! - Number of partial rounds = 59, where each partial round has one quintic s-box and (width-1) identity functions.
+//! 
+//! - Number of round constants = 960
+//! 
+//! ## Example outside R1CS
+//! ```
+//! use hades252::scalar::hash;
+//! use curve25519_dalek::scalar::Scalar;
+//! 
+//! // Generate the inputs that will be hashed.
+//! let input = [Scalar::one(); 7];
```

Comment:
Do you mean the value of the `Scalar`?

If you're talking about the length of the array used as input, the length of this array has to be `WIDTH -1` since we need one position at the beginning to add a `0`. So as max we could input: `[Scalar::one(); 8]`. 
But I don't think it matters too much since you can input a length that is less than `WIDTH - 1 == 8` and the `hash` function pads it for you.

Reply:
Yes, both will work, but maybe, as documentation, it is more explicit to state that we accept `WIDTH - 1` elements instead of `WIDTH - 2`. Or maybe we could put a comment that we arbitrarily decided to use `7`, but we could use up to `8`, to avoid the misinterpretation that we can put only `WIDTH - 2` elements.

What do you think?

Reply:
Yes, I'll do it with the maximum allowed length and say that we could call the function with fewer elements but never more.

Reply:
Indeed using 7 means that we use 8 inputs lol.

Everything should be fixed now.